### ISSUE TITLE
Application crashes when not connected after a long time

### ISSUE THREAD

**Issue by patrickelectric** [bluerobotics/ping-viewer#352]
```
#0  0x00007ffff6b65a80 in  () at /usr/lib/libQt5Qml.so.5
#1  0x00007ffff6b6c1fe in  () at /usr/lib/libQt5Qml.so.5
#2  0x00007ffff6b6c63b in  () at /usr/lib/libQt5Qml.so.5
#3  0x00007ffff6b6cbab in QQmlDelegateModel::_q_itemsInserted(int, int) () at /usr/lib/libQt5Qml.so.5
#4  0x00007ffff6b72af6 in  () at /usr/lib/libQt5Qml.so.5
#5  0x00007ffff6b72e83 in QQmlDelegateModel::qt_metacall(QMetaObject::Call, int, void**) () at /usr/lib/libQt5Qml.so.5
#6  0x00007ffff640ab4d in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#7  0x00007ffff6393f1f in QAbstractItemModel::rowsInserted(QModelIndex const&, int, int, QAbstractItemModel::QPrivateSignal) () at /usr/lib/libQt5Core.so.5
#8  0x00007ffff639ba0c in QAbstractItemModel::endInsertRows() () at /usr/lib/libQt5Core.so.5
#9  0x0000555555590151 in LogListModel::append(QString const&, QString const&, QColor const&, int) (this=0x5555556f6490, time=..., text=..., color=..., category=32768) at ../src/logger/loglistmodel.cpp:33
#10 0x000055555558db13 in Logger::logMessage(QString const&, QtMsgType const&, QMessageLogContext const&) (this=0x5555556f6450, msg=..., type=@0x7fffe2240604: QtDebugMsg, context=...) at ../src/logger/logger.cpp:54
#11 0x000055555558d66c in Logger::handleMessage(QtMsgType, QMessageLogContext const&, QString const&) (type=QtDebugMsg, context=..., msg=...) at ../src/logger/logger.cpp:71
#12 0x00007ffff621e8b6 in  () at /usr/lib/libQt5Core.so.5
#13 0x00007ffff622025d in qt_message_output(QtMsgType, QMessageLogContext const&, QString const&) () at /usr/lib/libQt5Core.so.5
#14 0x00007ffff6312031 in QDebug::~QDebug() () at /usr/lib/libQt5Core.so.5
#15 0x0000555555607743 in ProtocolDetector::checkSerial(LinkConfiguration&) (this=0x555555d2eca0, linkConf=...) at ../src/sensor/protocoldetector.cpp:99
#16 0x0000555555606cd3 in ProtocolDetector::checkLink(LinkConfiguration&) (this=0x555555d2eca0, linkConf=...) at ../src/sensor/protocoldetector.cpp:55
#17 0x0000555555606647 in ProtocolDetector::scan() (this=0x555555d2eca0) at ../src/sensor/protocoldetector.cpp:40
#18 0x000055555560e16b in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (ProtocolDetector::*)()>::call(void (ProtocolDetector::*)(), ProtocolDetector*, void**) (f=(void (ProtocolDetector::*)(ProtocolDetector * const)) 0x5555556064f0 <ProtocolDetector::scan()>, o=0x555555d2eca0, arg=0x7fffe2240dc0)
    at /usr/include/qt/QtCore/qobjectdefs_impl.h:134
#19 0x000055555560e0d3 in QtPrivate::FunctionPointer<void (ProtocolDetector::*)()>::call<QtPrivate::List<>, void>(void (ProtocolDetector::*)(), ProtocolDetector*, void**) (f=(void (ProtocolDetector::*)(ProtocolDetector * const)) 0x5555556064f0 <ProtocolDetector::scan()>, o=0x555555d2eca0, arg=0x7fffe2240dc0) at /usr/include/qt/QtCore/qobjectdefs_impl.h:167
#20 0x000055555560dff6 in QtPrivate::QSlotObject<void (ProtocolDetector::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) (which=1, this_=0x555555d2fca0, r=0x555555d2eca0, a=0x7fffe2240dc0, ret=0x0) at /usr/include/qt/QtCore/qobjectdefs_impl.h:396
#21 0x00007ffff640abd0 in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#22 0x00007ffff6227df8 in QThread::started(QThread::QPrivateSignal) () at /usr/lib/libQt5Core.so.5
#23 0x00007ffff6232f4c in  () at /usr/lib/libQt5Core.so.5
#24 0x00007ffff60c0a9d in start_thread () at /usr/lib/libpthread.so.0
#25 0x00007ffff5cc2b23 in clone () at /usr/lib/libc.so.6
```

**Comment by patrickelectric** [comment]
This issue appears to be fixed in Qt 5.12.
Tested and confirmed only in windows debug/release mode.

**Comment by patrickelectric** [comment]
This problem is still happening in Linux and MacOS.
